source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -91,7 +91,13 @@ absl::Status ProcessorState::handleHeadersResponse(const HeadersResponse& respon
         return mut_status;
       }
     }
-    if (common_response.clear_route_cache()) {
+
+    bool allow_clearing_route_cache =
+        PROTOBUF_GET_WRAPPED_OR_DEFAULT(filter_.config(), allow_clearing_route_cache, true);
+    // Only clear the route cache if there is a mutation to the filter.
```

Comment:
nit: The comment here needs to be updated.  Because in addition to the mutation to the header, we are also checking this new `allow_clearing_route_cache`

Reply:
Done.